extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -372,10 +378,17 @@ private void visitAnnotation(final DotName clazz, final IndexView index, final M
                         CDI_PRODUCES_ANNOTATION,
                         target,
                         Collections.emptyList()));
-                annotationsToAdd.add(create(
-                        CDI_DEPENDENT_ANNOTATION,
-                        target,
-                        Collections.emptyList()));
+                DotName declaredScope = getScope(methodInfo);
+                // Non-spring specific annotations are processed by Arc
+                if (declaredScope == null && !methodInfo.hasAnnotation(CDI_SINGLETON_ANNOTATION)) {
+                    declaredScope = CDI_SINGLETON_ANNOTATION; // implicit default scope in spring
+                }
+                if (declaredScope != null) {
```

Comment:
I don't think we ned this check anymore

Reply:
We need it for the case of explicit `@Singleton` scope. Otherwise the integration tests I've added trigger duplicate scopes error.
I find out that spring-di should not add `@Singleton` because Arc also does it (not sure where). This seems to make sense since @Singleton is not a spring thing ?